src/main/java/org/broadinstitute/dsde/consent/ontology/datause/services/TextTranslationServiceImpl.java

```diff
@@ -89,13 +98,67 @@ public String translateDataset(String dataUseString) throws IOException {
         return translate(dataUse, TranslateFor.DATASET);
     }
 
+    @Override
+    public Map<String, Recommendation> translateParagraph(String paragraph) throws IOException {
+        return paragraph(paragraph);
+    }
+
     @Override
     public String translatePurpose(String dataUseString) throws IOException {
         ObjectMapper mapper = new ObjectMapper();
         DataUse dataUse = mapper.readValue(dataUseString, DataUse.class);
         return translate(dataUse, TranslateFor.PURPOSE);
     }
 
+    private Map<String, Recommendation> paragraph(final String paragraph) throws IOException {
+      Map<String, Recommendation> recommendations = new HashMap<>();
+
+      List<TermItem> terms = loadTermsFromGoogleStorage();
+
+      if (terms.isEmpty()) {
+        throw new IOException("Unable to process search terms");
+      }
+
+      for (TermItem term : terms) {
+        final String title = term.getTitle();
+        final String category = term.getCategory();
+        final String url = term.getUrl();
+        final String[] keywords = term.getKeywords();
+
+        for (String keyword : keywords) {
+          final boolean foundMatch = searchForKeyword(keyword, paragraph);
+          if (foundMatch) {
+            recommendations.computeIfAbsent(url, key -> new Recommendation(
+                title,
+                category
+            ));
+            break;
+          }
+        }
+      }
+
+      return recommendations;
+    }
+
+    private static boolean searchForKeyword(final String keyword, final String targetText) {
+      return StringUtils.containsIgnoreCase(targetText, keyword);
```

Comment:
does this mean that if the paragraph contained, for example, the word `collection`, it would be tagged as containing `col`? Should this use regex to ensure that the keyword is surrounded by whitespace?

Reply:
@connorlbark - Yes,`keyword`: `col1` with `targetText`: `collection` would return true here.
@gabedemattos - I defer to you for the logical implementation. Is the intention here to find entire `keyword` word (with boundaries) within the target text or is it OK to to look for containing strings?

Reply:
I'm not fond of going with REGEX for this. In a future implementation we will switch to ElasticSearch, this is just a prototypical solution.